qutip/core/states.py

```diff
@@ -42,22 +43,23 @@ def _promote_to_zero_list(arg, length):
     raise TypeError("Dimensions must be an integer or list of integers.")
 
 
-def basis(dimensions, n=None, offset=None, *, dtype=_data.Dense):
+def basis(N, n=None, offset=None, *, dtype=_data.Dense):
```

Comment:
Here `dimensions` is a better name than `N`. It's more descriptive and I would expect `N` to only accept an integer.
`list of int` are the way we represent the dimension of the state.

Reply:
@Ericgig Ok so for functions that take int or list of int I should use dimensions and if it takes only int then N or should I change all to dimensions?